src/plugins/data/server/search/strategies/ese_search/request_utils.ts

```diff
@@ -46,37 +46,38 @@ export async function getDefaultAsyncSubmitParams(
     | 'keep_on_completion'
   >
 > {
+  // TODO: searchSessionsConfig could be "null" if we are running without x-pack which happens only in tests.
+  // This can be cleaned up when we completely stop separating basic and oss
+  const keepAlive = searchSessionsConfig?.enabled
+    ? `${searchSessionsConfig.defaultExpiration.asMilliseconds()}ms`
+    : '1m';
+
+  // Always return an ID for search sessions, even if the request completes quickly
+  const keepOnCompletion = searchSessionsConfig?.enabled && !!options.sessionId;
```

Comment:
Some of the settings here are incorrect.
Easier for me to fix than explain.
I will add comments.